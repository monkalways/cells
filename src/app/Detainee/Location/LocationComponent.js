import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography, withStyles } from '@material-ui/core';

import InCellIcon from '../../../images/InCell.png';
import InTransitIcon from '../../../images/InTransit.png';
import BailHearingGeneralIcon from '../../../images/BailHearingGeneral.png';
import FingerprintingIcon from '../../../images/Fingerprinting.png';
import MedicalVisitIcon from '../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../images/PhoneAccept.png';

const LOCATION_PHONE_ACCEPT = 'phone - in progress';
const LOCATION_MEDICAL_VISIT = 'medical - in progress';
const LOCATION_FINGER_PRINTING = 'fingerprinting - in progress';
const LOCATION_BAIL_HEARING = 'bail hearing - in progress';
const LOCATION_IN_TRANSIT = 'in transit';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  location: PropTypes.string.isRequired,
};

const LocationComponent = ({ classes, location }) => {
  const getLocationIconSrc = () => {
    let locationIconSrc = InCellIcon;

    if (location) {
      if (location.toLowerCase().includes(LOCATION_IN_TRANSIT)) {
        locationIconSrc = InTransitIcon;
      }

      switch (location.toLowerCase()) {
        case LOCATION_PHONE_ACCEPT:
          locationIconSrc = PhoneAcceptIcon;
          break;
        case LOCATION_MEDICAL_VISIT:
          locationIconSrc = MedicalVisitIcon;
          break;
        case LOCATION_FINGER_PRINTING:
          locationIconSrc = FingerprintingIcon;
          break;
        case LOCATION_BAIL_HEARING:
          locationIconSrc = BailHearingGeneralIcon;
          break;
        default:
          break;
      }
    }

    return locationIconSrc;
  };

  return (
    <Grid
      container
      direction="column"
      justify="space-between"
      className={classes.column}
      spacing={8}
    >
      <Grid item>
        <Typography>Location</Typography>
      </Grid>
      <Grid item>
        <img
          className={classes.img}
          alt="Detainee Location"
          src={getLocationIconSrc()}
        />
      </Grid>
    </Grid>
  );
};

LocationComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  img: {
    width: '150px',
    height: '150px',
  },
  column: {
    padding: theme.spacing.unit * 0.5,
  },
}))(LocationComponent);
