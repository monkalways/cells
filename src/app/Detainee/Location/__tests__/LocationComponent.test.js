import React from 'react';
import { shallow } from 'enzyme';

import { LocationComponent } from '../LocationComponent';

describe('LocationComponent', () => {
  let detainee;

  beforeEach(() => {
    detainee = {
      location: 'location',
    };
  });

  const setup = () => {
    const classes = {
      div: 'div',
      text: 'text',
    };
    return shallow(<LocationComponent classes={classes} detainee={detainee} />);
  };
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
