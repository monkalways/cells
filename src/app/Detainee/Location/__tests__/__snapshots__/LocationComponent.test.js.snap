// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationComponent should render 1`] = `
<WithStyles(Grid)
  container={true}
>
  <WithStyles(Grid)
    container={true}
    item={true}
    justify="center"
    xs={12}
  >
    <WithStyles(Typography)
      className="text"
      variant="h5"
    >
      Location
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    item={true}
    justify="center"
    xs={12}
  >
    <div
      className="div"
    >
      <WithStyles(LocationIconComponent)
        location="location"
      />
    </div>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
